pandas/_libs/parsers.pyx

```diff
@@ -1304,8 +1304,10 @@ cdef class TextReader:
             if self.header is not None:
                 j = i - self.leading_cols
                 # generate extra (bogus) headers if there are more columns than headers
+                # These should be strings, not integers, because otherwise we might get
+                # issues with callables as usecols GH#46997
```

Comment:
if these are truely bogus, why are they even passed to the usecols callable? There are no circumstances where these "auto-generated" column names are/should be in the result?

Reply:
see #47138, this is the bogus case


You can access them when using a index-based usecols setting, but the column name is shifted afterwards.

Reply:
> but the column name is shifted afterwards.

right, so if they don't currently make their way into the result, we are not at risk of changing existing behavior with this PR?

Reply:
No.

At least I can not see a way in how this would impact anything.

You can not select these columns by name via usecols. If you select them by position, an available name is used, not the bogus name.

The only impact this should have is consistensy when feeding them into the usecols callable

Reply:
> At least I can not see a way in how this would impact anything.

cool.